pkg/scheduler/eventhandlers.go

```diff
@@ -281,8 +312,10 @@ func addAllEventHandlers(
 			},
 		},
 	)
+	sched.RegisteredHandlers = append(sched.RegisteredHandlers, resourceEventHandlerRegistration)
```

Comment:
Append  `sched.RegisteredHandlers` again and again makes the code a bit redundant.  😓

Or we can change like  this? WDYT? 
```go
handles := []ResourceEventHandlerRegistration{}
.....
handlers = append(handles, reg)
......
sched.RegisteredHandlers = handlers
```

Reply:
Yea, it's a bit redundant, but I didn't find a clean way😢.  We can try your option first.